cli/module_generate/scripts/generate_stubs.py

```diff
@@ -3,15 +3,92 @@
 import subprocess
 import sys
 from importlib import import_module
+from typing import List, Set
 
 
-def return_attribute(resource_name: str, attr: str) -> ast.Attribute:
+def return_attribute(value: str, attr: str) -> ast.Attribute:
     return ast.Attribute(
-        value=ast.Name(id=resource_name, ctx=ast.Load()),
+        value=ast.Name(id=value, ctx=ast.Load()),
         attr=attr,
         ctx=ast.Load())
 
 
+def update_annotation(
+    resource_name: str,
+    annotation: ast.Name | ast.Subscript,
+    nodes: Set[str],
+    parent: str
+) -> ast.Attribute | ast.Subscript:
+    if isinstance(annotation, ast.Name) and annotation.id in nodes:
+        value = parent if parent else resource_name
+        return return_attribute(value, annotation.id)
+    elif isinstance(annotation, ast.Subscript):
+        annotation.slice = update_annotation(
+            resource_name,
+            annotation.slice,
+            nodes,
+            parent)
+    return annotation
+
+
+def replace_async_func(
+    resource_name: str,
+    func: ast.AsyncFunctionDef,
+    nodes: Set[str],
+    parent: str = ""
+) -> None:
+    for arg in func.args.args:
+        arg.annotation = update_annotation(
+            resource_name,
+            arg.annotation,
+            nodes,
+            parent)
+    func.body = [
+        ast.Raise(
+            exc=ast.Call(func=ast.Name(id='NotImplementedError',
+                                       ctx=ast.Load()),
+                         args=[], 
+                         keywords=[]),
+            cause=None)
+    ]
+    func.decorator_list = []
+    if isinstance(func.returns, (ast.Name, ast.Subscript)):
+        func.returns = update_annotation(
+            resource_name, func.returns, nodes, parent
+        )
+
+
+def return_subclass(
+        resource_name: str, stmt: ast.ClassDef, parent=""
```

Comment:
(minor) we're a bit inconsistent in this file with how we type annotate parameters. Probably we want `parent: str = ""`